From 3d97da608890e1986f5d370213512467c129fd25 Mon Sep 17 00:00:00 2001
From: Thomas Koch <linrunner@gmx.net>
Date: Tue, 2 Jan 2018 18:02:04 +0100
Subject: [PATCH] Issue #313: don't detect wireless input devices' batteries as
 power supply

Rationale: wireless input devices - e.g. Logitech mice and
keyboards - expose their batteries below /sys/class/power_supply
with recent kernels.

Solution: ignore /sys/class/power_supply/hidpp_battery*.

Refererence:
* Issue #313: https://github.com/linrunner/TLP/issues/313
---
diff --git a/tlp-functions.in b/tlp-functions.in
index ab7aae2..3fe708f 100644
--- a/tlp-functions.in
+++ b/tlp-functions.in
@@ -348,12 +348,15 @@ get_sys_power_supply () { # get current power supply
                 ;;
 
             Battery)
-                # set rc to battery, but don't stop looking for AC
-                syspwr=1
+                # battery detected, exclude hid peripherals
+                if echo "${psrc##/*/}" | grep -vq "hidpp_battery"; then
+                    # not hid --> set rc to battery, but don't stop looking for AC
+                    syspwr=1
+                fi
                 ;;
 
             *)
-                echo_debug "pm" "unknown power supply: ${psrc##*/}"
+                echo_debug "pm" "unknown power supply: ${psrc##/*/}"
                 ;;
         esac
     done
